internal/appconfig/validation.go

```diff
@@ -158,6 +158,11 @@ func (cfg *Config) validateDeploySection() (extraInfo string, err error) {
 			extraInfo += "error canary deployment strategy is not supported when using mounted volumes"
 			err = ValidationError
 		}
+
+		if s != "rolling" && cfg.Deploy.MaxUnavailable != 0.0 {
```

Comment:
I wouldn't make it so strict, just ignore the field if not rolling or show a warning. But a validation error is too much. My personal take though.

Reply:
initially I couldn't see a way this could happen that wasn't an accident, but looking at it again there are places where it'd be helpful to have that flexibility.